dogfood/contents/Dockerfile

```diff
@@ -2,27 +2,29 @@ FROM rust:slim AS rust-utils
 # Install rust helper programs
 # ENV CARGO_NET_GIT_FETCH_WITH_CLI=true
 ENV CARGO_INSTALL_ROOT=/tmp/
-RUN cargo install exa bat ripgrep typos-cli watchexec-cli
+RUN cargo install exa bat ripgrep typos-cli watchexec-cli && \
+	# Reduce image size.
+	rm -rf /usr/local/cargo/registry
 
 FROM ubuntu:jammy AS go
 
-RUN apt-get update && apt-get install --yes curl gcc
 # Install Go manually, so that we can control the version
 ARG GO_VERSION=1.22.5
-RUN mkdir --parents /usr/local/go
 
 # Boring Go is needed to build FIPS-compliant binaries.
-RUN curl --silent --show-error --location \
+RUN apt-get update && \
+	apt-get install --yes curl && \
+	curl --silent --show-error --location \
 	"https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz" \
 	-o /usr/local/go.tar.gz
 
-RUN tar --extract --gzip --directory=/usr/local/go --file=/usr/local/go.tar.gz --strip-components=1
-
 ENV PATH=$PATH:/usr/local/go/bin
-
-# Install Go utilities.
 ARG GOPATH="/tmp/"
-RUN mkdir --parents "$GOPATH" && \
+# Install Go utilities.
+RUN apt-get install --yes gcc && \
```

Comment:
You should still just merge the gcc with the curl apt install above to save build time

Reply:
It can help cache optimization for when a Go tool version changes, but beyond that I don't see how there would be an improvement in build time?

Installing and uninstalling is a significant size optimization though:

```
sha256:ef2ad8c02872f01a9e71fc56afdf2ecda2a3d827ac61f5a9e1be0b096b79c335   About a minute ago   RUN /bin/sh -c apt-get install gcc --yes # buildkit                                                 159MB     buildkit.dockerfile.v0
```

Reply:
This is install+cleanup:
```
sha256:b39c284ff9fac0da7a256f29de3116c5109f4f04e12291ded468314dde50b28a   19 seconds ago   RUN /bin/sh -c apt-get install gcc --yes && apt-get remove --yes gcc && apt-get autoremove --yes && apt-get clean # buildkit   1.64MB    buildkit.dockerfile.v0
```

I thought `apt-get update` was pretty innocent but it's actually 50MB too. 😅